### PR TITLE
Data downloads

### LINKED ISSUE #25: Allow Subject Data Downloads
Hook up functionality to download subject data. According to the frontend, pressing this button will download a .zip file containing
- line by line transcription and metadata (.csv)
- text-only file (.txt)
- Raw, unparsed transcription data (.json)

### PR DESCRIPTION
**Pull request by wgranger**
Closes #25 

Hooks up the UI to allow downloading group exports on the [transcriptions page](https://projects.invisionapp.com/d/#/console/17925240/375544496/preview) and single transcriptions on the [edit page](https://projects.invisionapp.com/d/#/console/17925240/371813753/preview).

- Downloading subject set data should download all approved subjects in a subject group into a .zip.
- Downloading a single subject from the edit page will also download a .zip for that individual subject.

**Note**: Transcription must be marked "approved" to download export

### CODE REVIEW

**Comment by wgranger** [comment]
@chelseatroy Some notes, as this is your first PR reviewing. Once you log in to staging, you should click on Anti-Slavery Testing, then the only workflow you see, then either the "12345" group to see some valid transcriptions you can interact with. All of the other paths will lead you to mock data with limited interactivity.

**Comment by chelseatroy** [comment]
> @chelseatroy Some notes, as this is your first PR reviewing. Once you log in to staging, you should click on Anti-Slavery Testing, then the only workflow you see, then either the "12345" group to see some valid transcriptions you can interact with. All of the other paths will lead you to mock data with limited interactivity.

Cool. What specifically should work from this PR? Can you please provide screenshot examples of what the .zips should look like when expanded?

**Comment by wgranger** [comment]
Do download a single transcription .zip

First, make sure the subject is approved so you are able to download it:
approve

Second, click on the "More" button to view the option to download data:
download

Your .zip should look like this:
zip

**Comment by wgranger** [comment]
To download a group transcription set:

First, visit the page with all transcriptions for a group and click on "download set data":
Screen Shot 2020-03-16 at 3 08 50 PM

The zip should look like this:
Screen Shot 2020-03-16 at 3 09 20 PM

**Comment by chelseatroy** [comment]
Sweet. The functionality works as described on my machine. Nice work @wgranger!

Now I'll dive into the code. Please hold ⏲

File: src/components/Modals/DownloadDataModal/DownloadDataModalContainer.spec.js [hunk 1]
```diff
@@ -30,4 +36,10 @@ describe('Component > DownloadDataModalContainer', function () {
     wrapper.props().onClose()
     expect(toggleModalSpy).toHaveBeenCalledWith('')
   })
+
+  it('should call toggleModal on child onClose', function () {
```
Comment (chelseatroy) [inline src/components/Modals/DownloadDataModal/DownloadDataModalContainer.spec.js]: It looks like this test might have been duped from the one above it. The test names are identical and there's a duped assertion (line 37 and line 43 are identical).

Can we change the name of this test to 'should call dataDownload when download clicked'?

Might also not need the assertion on line 43 (unless you also want to verify that the modal closes when you click download button).

File: src/store/ClientStore.js [hunk 2]
```diff
@@ -15,10 +17,38 @@ const ClientStore = types.model('ClientStore', {
...
+      : `/transcriptions/${currentTranscription}/export`
+    const filename = isEntireGroup ? `${currentGroup}_export` : `${currentTranscription}_export`
+
+      try {
+        yield self.toveZip.get(query).then(response => response.blob())
+          .then(blob => download(blob, `${filename}.zip`))
+      } catch (error) {
+        console.log(error);
+      }
+  }),
```
Comment (chelseatroy) [inline src/store/ClientStore.js]: Nice. Very clean :)

File: src/components/Modals/DownloadDataModal/DownloadDataModalContainer.spec.js (hunk 1 above)
Comment (wgranger) [reply src/components/Modals/DownloadDataModal/DownloadDataModalContainer.spec.js]: Good catch! Yes, I'd also like to assert the modal closes when the download button is clicked.

Comment (wgranger) [reply src/components/Modals/DownloadDataModal/DownloadDataModalContainer.spec.js]: The duped assertion is on purpose, as that is to check that the modal closes.

**Comment by wgranger** [comment]
Thanks for the review! I adjusted the test description and will merge after checks pass.

**Pull request merged**